src/reducers/vis-state-updaters.js

```diff
@@ -582,15 +573,10 @@ export function setFilterUpdater(state, action) {
   // filter data
   newState = {
     ...newState,
-    datasets: {
-      ...newState.datasets,
-      [dataId]: {
-        ...newState.datasets[dataId],
-        ...filterData(allData, dataId, newState.filters)
-      }
-    }
+    datasets: applyFiltersToDatasets(dataId, newState.datasets, newState.filters)
   };
 
+  // dataId is an array
   newState = updateAllLayerDomainData(newState, dataId, newFilter);
```

Comment:
if a filter is applied to dataId ['a', 'b'], and only the field name of dataset 'a' is updated. you don't neet to filter both 'a' and 'b'. You should only apply filter to dataset at `datasetIdx`